src/client/thingtypemanager.cpp

```diff
@@ -211,6 +212,89 @@ bool ThingTypeManager::loadAppearances(const std::string& file)
     }
 }
 
+namespace {
+    using RaceBank = google::protobuf::RepeatedPtrField<staticdata::Creature>;
+    
+    void loadCreatureBank(RaceList& otcRaceList, const RaceBank& protobufRaceList, bool boss) {
+        for (const auto& protobufRace : protobufRaceList) {
+            // add race to vector
+            RaceType otcRaceType = RaceType();
+            otcRaceType.raceId = protobufRace.raceid();
+            otcRaceType.name = protobufRace.name();
+            otcRaceType.boss = boss;
+
+            Outfit otcOutfit;
+            const auto& protobufOutfit = protobufRace.outfit();
+            if (protobufOutfit.lookitem() != 0) {
+                otcOutfit.setAuxId(static_cast<uint16_t>(protobufOutfit.lookitem()));
+            } else {
+                otcOutfit.setId(static_cast<uint16_t>(protobufOutfit.looktype()));
+                otcOutfit.setAddons(static_cast<uint8_t>(protobufOutfit.lookaddons()));
+                if (protobufOutfit.has_colors()) {
+                    const auto& pbColors = protobufOutfit.colors();
+                    otcOutfit.setHead(static_cast<uint8_t>(pbColors.head()));
+                    otcOutfit.setBody(static_cast<uint8_t>(pbColors.body()));
+                    otcOutfit.setLegs(static_cast<uint8_t>(pbColors.legs()));
+                    otcOutfit.setFeet(static_cast<uint8_t>(pbColors.feet()));
+                }
+            }
+
+            otcRaceType.outfit = otcOutfit;
+            otcRaceList.emplace_back(otcRaceType);
+        }
+    }
+}
+
+bool ThingTypeManager::loadStaticData(const std::string& file)
+{
+    try {
+        std::string staticDataFile;
+
+        json document = json::parse(g_resources.readFileContents(g_resources.resolvePath(g_resources.guessFilePath(file + "catalog-content", "json"))));
+        for (const auto& obj : document) {
+            const auto& type = obj["type"];
+            if (type == "staticdata") {
+                staticDataFile = obj["file"];
+            }
+        }
+
+        // load staticdata.dat
+        std::stringstream datFileStream;
+        g_resources.readFileStream(g_resources.resolvePath(stdext::format("%s%s", file, staticDataFile)), datFileStream);
+        auto staticDataLib = staticdata::Staticdata();
+        if (!staticDataLib.ParseFromIstream(&datFileStream)) {
+            throw stdext::exception("Couldn't parse staticdata lib.");
+        }
+
+        // if reload, start again
+        m_monsterRaces.clear();
+
+        const auto& raceBank = staticDataLib.monsters();
+        const auto& bossBank = staticDataLib.bosses();
+        //m_monsterRaces.resize(static_cast<size_t>(raceBank.size()) + bossBank.size());
```

Comment:
You might want to use "reserve" to avoid reallocations and improve performance.

Reply:
I was close, just forgot the function name😅